tempdir.go

```diff
@@ -0,0 +1,91 @@
+package ibctest
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+)
+
+// TempDirTestingT is a subset of testing.TB to implement TempDir.
+type TempDirTestingT interface {
+	Helper()
+
+	Name() string
+
+	Failed() bool
+	Cleanup(func())
+
+	Logf(format string, args ...interface{})
+	Errorf(format string, args ...interface{})
+}
+
+// KeepTempDirOnFailure determines whether a directory created by TempDir
+// is retained or deleted following a test failure.
+//
+// It defaults to false, but can be initialized to true by setting the
+// environment variable IBCTEST_KEEP_FAILURE_DIR to a non-empty value.
+// Alternatively, importers of the ibctest package may set the variable to true.
+var KeepTempDirOnFailure = os.Getenv("IBCTEST_KEEP_FAILURE_DIR") != ""
```

Comment:
I'm open to suggestions on a better name for this environment variable.

The `IBCTEST_` prefix would be standard. `KEEP_FAILURE_DIR` reads a little like "the directory in which failures should be kept", but matching the whole name `IBCTEST_KEEP_TEMP_DIR_ON_FAILURE` is a bit too long IMO.

Reply:
`KEEP_DATA_ON_FAILURE`, `KEEP_DIR_ON_FAILURE`?

I also can't think of a shorter name that's as clear. 

---

Another note: Should we add this env var to the README or assume godoc is good enough?